llvm/utils/TableGen/GlobalISel/PatternParser.cpp

```diff
@@ -0,0 +1,462 @@
+//===- PatternParser.cpp ----------------------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "PatternParser.h"
+#include "../CodeGenIntrinsics.h"
+#include "../CodeGenTarget.h"
```

Comment:
Avoid using relative include paths

Reply:
They're not avoidable here, `Pattern.cpp` also needs them

Reply:
don't you just need to add the TableGen top level dir to the include directories?